packages/@icon-magic/distribute/src/index.ts

```diff
@@ -61,16 +56,17 @@ export async function distributeByType(
         break;
       }
       case 'svg': {
-        await distributeSvg(icon, outputPath);
+        await distributeSvg(icon, outputPath, groupByCategory, spriteNames);
         break;
       }
       default: {
         await createImageSet(icon, outputPath);
         await distributeByResolution(icon, outputPath);
-        await distributeSvg(icon, outputPath);
+        await distributeSvg(icon, outputPath, groupByCategory, spriteNames);
       }
     }
   }
+  if (type === 'svg') await writeSpriteToFile(spriteNames, outputPath);
```

Comment:
Move this into `distributeSvg`. This also means moving the iconset loop to within the individual type functions. Also, we'll need to support multiple types passed in as options, so the switch case will need to be updated.